src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -173,6 +181,14 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
     }
   }, [adapter, wallet])
 
+  useEffect(() => {
+    if (!accountId) return
+    ;(async () => {
+      const accountAddress = fromAccountId(accountId).account
+      setConnectedWalletEthAddress(accountAddress)
+    })()
+  }, [accountId])
+
```

Comment:
This is the guts of this PR - won't bring any changes in for now.

Note there is no change of terminology for now as the address *is* the connected wallet address until we have more accounts than `account[0]`, will be tackled in a follow-up PR, with a final one to remove the hardcoding of account zero.